app/src/main/kotlin/com/wire/android/ui/authentication/welcome/Welcome.kt

```diff
@@ -80,50 +82,64 @@ private fun WelcomeContent(navController: NavController) {
                 WelcomeCarousel()
             }
 
-            Column(modifier = Modifier.padding(top = 40.dp, bottom = 52.dp, start = 16.dp, end = 16.dp),) {
+            Column(modifier = Modifier.padding(
+                vertical = MaterialTheme.wireDimensions.welcomeVerticalSpacing,
+                horizontal = MaterialTheme.wireDimensions.welcomeButtonHorizontalPadding
+            )) {
                 LoginButton {
                     navController.navigate(AuthDestination.loginScreen)
                 }
                 CreateEnterpriseAccountButton {
                     navController.navigate((AuthDestination.createEnterpriseAccount))
-
                 }
             }
 
-            WelcomeFooter(modifier = Modifier.padding(bottom = 56.dp, start = 16.dp, end = 16.dp),
+            WelcomeFooter(modifier = Modifier
+                .padding(horizontal = MaterialTheme.wireDimensions.welcomeTextHorizontalPadding),
                 onPrivateAccountClick = {
                     navController.navigate(AuthDestination.createPrivateAccountScreen)
                 })
         }
     }
 }
 
-@OptIn(ExperimentalPagerApi::class, ExperimentalFoundationApi::class)
+@OptIn(ExperimentalPagerApi::class, ExperimentalFoundationApi::class, ExperimentalCoroutinesApi::class)
 @Composable
 private fun WelcomeCarousel() {
     val delay = integerResource(id = R.integer.welcome_carousel_item_time_ms)
     val icons: List<Int> = typedArrayResource(id = R.array.welcome_carousel_icons).drawableResIdList()
     val texts: List<String> = stringArrayResource(id = R.array.welcome_carousel_texts).toList()
     val items: List<Pair<Int, String>> = icons zip texts
-    val circularItemsList = listOf<Pair<Int, String>>().plus(items.last()).plus(items)
-    val pageState = rememberPagerState(initialPage = 1)
-
-    LaunchedEffect(pageState.currentPage) {
-        if (pageState.currentPage == circularItemsList.lastIndex) {
-            pageState.scrollToPage(0)
-        } else {
-            delay(delay.toLong())
-            pageState.animateScrollToPage(pageState.currentPage + 1)
-        }
+    val circularItemsList = listOf<Pair<Int, String>>().plus(items.last()).plus(items).plus(items.first())
+    val initialPage = 1
+    val pageState = rememberPagerState(initialPage = initialPage)
+
+    LaunchedEffect(pageState) {
```

Comment:
also while reading, not really sure what this LaunchedEffect is doing, maybe because I am newbie with LaunchedEffect :P but maybe it would be a good idea to creata a private function with a name that clarifies what is happening ?